src/main/java/org/ice4j/ice/Component.java

```diff
@@ -169,10 +171,12 @@
      * @param componentID the id of this component.
      * @param mediaStream the {@link IceMediaStream} instance that would be the
      * parent of this component.
+     * @param useComponentSocket whether the component socket should be used.
      */
-    protected Component(int            componentID,
+    protected Component(int componentID,
                         IceMediaStream mediaStream,
-                        KeepAliveStrategy keepAliveStrategy)
+                        KeepAliveStrategy keepAliveStrategy,
+                        boolean useComponentSocket)
```

Comment:
Same here with keeping the existing constructor and the version.

Reply:
This one is `protected`. Theoretically, people might extend `Component`, but I highly doubt anyone is doing it. Are there any automated tools that check the API compatibility? Do you still think it's necessary to re-add the constructor with the previous signature?

Reply:
I added it, it doesn't hurt

Reply:
Part of the Maven build should actually check the API compatibility.